clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -704,11 +704,14 @@ public void sendOffsetsToTransaction(Map<TopicPartition, OffsetAndMetadata> offs
         throwIfInvalidGroupMetadata(groupMetadata);
         throwIfNoTransactionManager();
         throwIfProducerClosed();
-        long start = time.nanoseconds();
-        TransactionalRequestResult result = transactionManager.sendOffsetsToTransaction(offsets, groupMetadata);
-        sender.wakeup();
-        result.await(maxBlockTimeMs, TimeUnit.MILLISECONDS);
-        producerMetrics.recordSendOffsets(time.nanoseconds() - start);
+
+        if (offsets.isEmpty()) {
```

Comment:
This check should be inverted.

Reply:
Ah my bad..

I was trying to fix both on the caller (streams) side as well as the callee (producer) side, but the callee part was added in a last minute and obviously I need some coffee at that time already.